Lib/test/test_email/test_headerregistry.py

```diff
@@ -1224,7 +1244,7 @@ def test_simple_address_list(self):
             'Harry W. Hastings')
 
     def test_complex_address_list(self):
-        examples = list(self.example_params.values())
+        examples = list(self.example_params.values())[:-1]
```

Comment:
This change is not future proof. What if someone adds new tests? In fact, you are just hidding that `name_starting_with_dot` is not passing the test:

    ERROR: test_complex_address_list (test.test_email.test_headerregistry.TestAddressHeader)
    ----------------------------------------------------------------------
    Traceback (most recent call last):
    File "Lib/email/_header_value_parser.py", line 1907, in get_address
        token, value = get_group(value)
    File "Lib/email/_header_value_parser.py", line 1866, in get_group
        "display name but found '{}'".format(value))
    email.errors.HeaderParseError: expected ':' at end of group display name but found '<jxd@example.com>'

    During handling of the above exception, another exception occurred:

    Traceback (most recent call last):
    File "Lib/test/test_email/test_headerregistry.py", line 1284, in test_complex_address_list
        h = self.make_header('to', source)
    File "Lib/test/test_email/test_headerregistry.py", line 78, in make_header
        return self.factory(name, value)
    File "Lib/email/headerregistry.py", line 589, in __call__
        return self[name](name, value)
    File "Lib/email/headerregistry.py", line 197, in __new__
        cls.parse(value, kwds)
    File "Lib/email/headerregistry.py", line 340, in parse
        kwds['parse_tree'] = address_list = cls.value_parser(value)
    File "Lib/email/headerregistry.py", line 331, in value_parser
        address_list, value = parser.get_address_list(value)
    File "Lib/email/_header_value_parser.py", line 1930, in get_address_list
        token, value = get_address(value)
    File "Lib/email/_header_value_parser.py", line 1910, in get_address
        token, value = get_mailbox(value)
    File "Lib/email/_header_value_parser.py", line 1733, in get_mailbox
        token, value = get_name_addr(value)
    File "Lib/email/_header_value_parser.py", line 1716, in get_name_addr
        token[0][:0] = [leader]
    TypeError: 'ValueTerminal' object does not support item assignment

I think you should try to fix this one too :)